app/javascript/mastodon/main.js

```diff
@@ -27,6 +27,12 @@ function main() {
   // import customization styles
   require.context('../../assets/stylesheets/', false, /custom.*\.scss$/);
 
+  if (process.env.NODE_ENV === 'production') {
+    // avoid offline in dev mode because it's harder to debug
+    const OfflinePluginRuntime = require('offline-plugin/runtime');
+    OfflinePluginRuntime.install();
```

Comment:
Unless `offline-plugin` is running a lot of code on the main thread, there's no reason I can see to delay it. Even if that's the case, I think `requestIdleCallback()` would be a better choice.

And yeah @NekR most Mastodon instances should be on http2.